modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/authcode/UpdatingAuthorizationCodeState.java

```diff
@@ -55,14 +60,22 @@ private void update(ResourceOwnerOAuthContext context) {
         onUpdate.accept(context);
       }
     });
+    this.dancer = dancer;
+    this.config = config;
+    this.muleContext = muleContext;
   }
 
   @Override
   public String getAccessToken() {
-    if (invalidated) {
-      throw new TokenInvalidatedException(
-                                          "OAuth token for resource owner id " + delegate.getResourceOwnerId()
-                                              + " has been invalidated");
+    if (invalidated || !muleContext.getClusterId().isEmpty()) {
```

Comment:
I don't think we should rely on the muleContext for this. Not only because of our long term goal to reduce its exposure and usage, but also because I think it'd be best to have the `ClusterService` provide this information. Receiving the ClusterService instead of the MuleContext also makes the code way more clear and self-documenting, allowing future developers to quickly understand the Cluster is a factor, vs simply passing the muleContext for an infinity of possible reasons.

I think it'd be best to have something like `if (clusterService.isClusterEnabled())` or something like that, rather than coupling to the clusterId concept which should be more internal.

Reply:
After the constructor AuthorizationCodeConnectionProviderWrapper, we can get information from getClusterId(). It will be empty when the instance does not belong to a cluster, based on NullClusterConfiguration. We are using the boolean isClusterEnabled here.